docs/user-guide/utils.rst

```diff
@@ -4,15 +4,83 @@
 Utilities
 *********
 
-Overview
-========
 This package provides a number of utilities to work with CCSDS packets.
 
+.. contents::
+   :depth: 2
+
+Iterating through Packet Bytes
+==============================
+During debugging it is often useful to break a stream of multiple packets into a list of byte sequences associated with each packet. That is, the i'th element of the list will be a `bytes` object assosciated with the i'th packet in the file.
+
+This can be done with the `utils.split_packet_bytes()` (returns a list) and `utils.iter_packet_bytes()` (returns a generator) functions. These functions have an optional keyword arugment `include_primary_header=True` which determines whether the primary header is included in the byte sequence returned. By default, it is included.
+
+This function works with mixed files containing multiple APIDs, which may include both fixed length and variable length packets.
+
+.. code-block:: python
+
+  from ccsdspy.utils import split_packet_bytes
+
+  packet_bytes = split_packet_bytes('mixed_file.bin')
+
+  # Print bytes of first five packets
+  for i in range(5):
+    print(f"The bytes in packet number {i+1} out "
+          f"of {len(packet_bytes)} are: ")
+    print(packet_bytes[i].hex())
+
+  print()
+  print("Stopping after first five!")
+
+
+This generates the following output for a sequence of variable length packets:
+
+.. code-block::
+
+  The bytes in packet number 1 out of 10 are:
+  20e20000000300020003
+  The bytes in packet number 2 out of 10 are:
+  20e200010005000300040005
+  The bytes in packet number 3 out of 10 are:
+  20e20002000900050006000700080009
+  The bytes in packet number 4 out of 10 are:
+  20e20003000d000700080009000a000b000c000d
+  The bytes in packet number 5 out of 10 are:
+  20e200040015000b000c000d000e000f001000110012001300140015
+
+  Stopping after first five!
+
+
+Counting Number of Packets in a File
+====================================
+Sometimes, it is desirable to know  the number of packets in a file. For this, the `utils.count_packets()` function can be used. It's argument is a file-like object or name of a file. It accepts the optional arguments `return_missing_bytes=True` which can be used to determine the number of bytes which would be needed to be added to the file to complete the last packet (computed using the packet length set in the primary header of last packet).
+
+This function works with mixed files containing multiple APIDs, which may include both fixed length and variable length packets.
```

Comment:
How does it work if there are multiple APIDs? Does it give you the number of packets per APID?

Reply:
It just counts the number of packets of any APIDs. You could mix this with split_by_apid() if you wanted to count per APID. I added a comment to the doc